### DISCUSSION TITLE
KaTeX Macros

### DISCUSSION THREAD

**Discussion by fweth** [facebook/docusaurus#10270] [General] [answered]
Are [KaTeX macros](https://katex.org/docs/supported#macros) supported by the [recommended](https://docusaurus.io/docs/markdown-features/math-equations) KaTeX pipeline?  I couldn't get them to work, i.e. my markdown code which works fine in the VSCode preview (which also uses KaTeX) doesn't work with Docusaurus.  But maybe that's an issue of the remark/rehype plugins?

Just to make clear what I'm talking about, I have a section like this at the top of my markdown files

```latex
$$
\def\cat{\mathsf{Cat}}
\def\hom{\mathsf{Hom}}
\def\id{\mathsf{id}}
\def\psh{\mathsf{PSh}}
\def\set{\mathsf{Set}}
\def\dot{\ *\ }
\def\op{\mathsf{op}}
\def\rr{\mathbb{R}}
\def\zz{\mathbb{Z}}
$$
```
and then I use those abbreviations throughout the document.  Is this somehow possible with Docusaurus?

Just to make clear, you can use macros inside a single math block just fine.  But the global context between math blocks seems lacking.

**Comment by fweth** [comment]
Apparently global math context is not supported by `rehype-katex`, but macros can be supplied as options to the `rehype` plugin, can I somehow supply those options via Docusaurus as well?  Like

```js
rehypePlugins: [{plugin: rehypeKatex, options: {macros: {"\\RR": "\\mathbb{R}"}}],
```
except that this doesn't work?

**Reply by Josh-Cena** [reply, marked as answer]
https://docusaurus.io/docs/markdown-features/plugins#configuring-plugins
